WebDriverAgentLib/Utilities/FBSearchPredicatesFormatter.m

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "FBSearchPredicatesFormatter.h"
+#import "FBElement.h"
+#import "FBElementUtils.h"
+
+@implementation NSPredicate (ExtractComparisons)
+
+- (NSPredicate *)predicateByChangingComparisonsWithBlock:(NSPredicate *(^)(NSComparisonPredicate *))block {
```

Comment:
1. This is category on class you do not own so you should prefix it with `fb_`
2. I would make it a class method as a constructor e.g. `+(instancetype)predicateWithPredicate: comparisonModifier:`